### PR TITLE
Final multimodal PR with our recent developments on MM side

### PR DESCRIPTION
**Pull request by yaoyu-33**
# What does this PR do ?

multimodal PR with our recent developments on MM side after we start the open sourcing process. Mainly update in Neva/Llava and stable diffusion

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Jenkins CI
To run Jenkins, a NeMo User with write access must comment `jenkins` on the PR.

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by yaoyu-33** [comment]
jenkins

**Comment by yaoyu-33** [comment]
jenkins

**Comment by yaoyu-33** [comment]
jenkins

**Comment by yaoyu-33** [comment]
jenkins

**Comment by yaoyu-33** [comment]
jenkins

File: examples/multimodal/multimodal_llm/neva/eval/gardio_server.py
Comment (ericharper) [inline examples/multimodal/multimodal_llm/neva/eval/gardio_server.py:1]: can we change this filename to gradio_server.py ?

File: nemo/collections/multimodal/data/neva/neva_dataset.py
Comment (ericharper) [inline nemo/collections/multimodal/data/neva/neva_dataset.py:1]: Why are the docstrings being removed from this file?

If we keep them we should fix their formatting.

File: nemo/collections/multimodal/parts/stable_diffusion/lr_scheduler.py
Comment (ericharper) [inline nemo/collections/multimodal/parts/stable_diffusion/lr_scheduler.py:1]: can probably just add these to nemo schedulers (can be a later PR)

File: nemo/collections/nlp/models/language_modeling/megatron_base_model.py
Comment (ericharper) [inline nemo/collections/nlp/models/language_modeling/megatron_base_model.py:1]: can we put these changes in it's own PR, a lot of models depend on this class

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py
Comment (ericharper) [inline nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py:1]: Same for this class, can we put changes in a separate PR?

File: nemo/collections/nlp/modules/common/text_generation_utils.py [hunk 1]
```diff
@@ -148,10 +149,12 @@ def megatron_neva_generate(model, prompt_dict_list, length_params, sampling_para
     conv_template = model.cfg.data.get("conv_template", "nvgpt")
     final_response = []
     for idx, prompt_dict in enumerate(prompt_dict_list):
-        img = os.path.join(inference_config.inference.images_base_path, prompt_dict['image'])
+        import pdb
+
+        pdb.set_trace()
```
Comment (ericharper) [inline nemo/collections/nlp/modules/common/text_generation_utils.py:154]: remove debug statement

File: nemo/collections/vision/modules/common/megatron/vision_transformer.py
Comment (ericharper) [inline nemo/collections/vision/modules/common/megatron/vision_transformer.py:1]: just commenting here that this should be updated to use mcore

File: nemo/collections/multimodal/parts/stable_diffusion/lr_scheduler.py
Comment (yaoyu-33) [reply nemo/collections/multimodal/parts/stable_diffusion/lr_scheduler.py:1]: ok

**Comment by yaoyu-33** [comment]
jenkins

File: nemo/collections/vision/modules/common/megatron/vision_transformer.py
Comment (yaoyu-33) [reply nemo/collections/vision/modules/common/megatron/vision_transformer.py:1]: yes, i will move vit to mcore

**Comment by ericharper** [comment]
jenkins

**Comment by ericharper** [comment]
jenkins

**Comment by yaoyu-33** [comment]
jenkins

**Comment by yaoyu-33** [comment]
jenkins

**Review by ericharper — APPROVED** [review]
LGTM. Thanks!
